dirty_equals/_sequence.py

```diff
@@ -87,7 +87,9 @@ class IsListOrTuple(DirtyEquals[T]):
     Check that some object is a list or tuple and optionally its values match some constraints.
     """
 
-    allowed_type: Union[Type[T], Tuple[Type[List[Any]], Type[Tuple[Any, ...]]]] = (list, tuple)
+    allowed_type: Union[
+        Tuple[Type[List[Any]]], Tuple[Type[Tuple[Any, ...]]], Tuple[Type[List[Any]], Type[Tuple[Any, ...]]]
+    ] = (list, tuple)
```

Comment:
This is a hack to be able to perform: `' or '.join(t.__name__ for t in self.allowed_type)`

`allowed_type` is used only in `if not isinstance(other, self.allowed_type):` which does not change result for

```diff
- if not isinstance(other, list):
+ if not isinstance(other, (list, )):
```